src/persistence/db/upgrades/dbupgrader.cpp

```diff
@@ -285,13 +287,15 @@ bool DbUpgrader::createCurrentSchema(RawDatabase& db)
 {
     QVector<RawDatabase::Query> queries;
```

Comment:
I like the idea of db upgrader, and I know that this isn't introduced in this commit, but I think having createCurrentSchema in history.cpp was a lot easier to follow. Not really requesting a change, just making a note.

Reply:
Hm, I think it's nice to keep the general version handling in dbupgrader - and dbupgrader depending on history to get the current schema feels a bit weird?

I guess history could handle just `isNewDb` and `createCurrentSchema`, then dispatch to DbUpgrader::dbSchemaUpgrade otherwise.

It would also mean that tests which currently just depend on dbupgrader would depend on history as well to get the current schema made.

Reply:
It's fine the way it is, I just thought I'd note it.